Automattic/wp-calypso
# Site Assembler: Improve the global styles gating experience

Related to https://github.com/Automattic/wp-calypso/issues/74946

## Proposed Changes

* Updated copy in the modal
  * Added `You've selected a premium color or font for your site.` as the first sentence.
  * Added `You can also continue with your selected color or font and upgrade later."` as the last sentence.
  * Replaced `Included with your purchase` with `Included with your Premium plan`
  * Replaced `Try it out` with `Upgrade later` and have it close the modal
* When the modal closes
  * The button in the side nav changes to `Continue`
  * The description above the button changes to `Ready? Keep your styles and go to the Site Editor to edit your content. You’ll be able to upgrade to the Premium plan later.`

### Screenshots

![image](https://user-images.githubusercontent.com/13596067/228488416-095fe187-0c4e-41c2-8bdc-c76d51dd3b25.png)

![image](https://user-images.githubusercontent.com/13596067/228488967-8cf4804b-4ece-450d-8629-c827b4dc5682.png)

![Screenshot 2023-03-29 at 5 21 09 PM](https://user-images.githubusercontent.com/13596067/228488901-e74c737a-e467-431e-b80e-cf43a067d5fc.png)

### Demo

https://user-images.githubusercontent.com/13596067/228487833-e67c6052-4f03-4287-9e5c-47ca0fd53b95.mov

## Testing Instructions



* Go to /setup/site-setup?siteSlug=<your_site>
  * Note that the site should be the site with a free plan
* Continue until you land on the Design Picker
* On the Design Picker screen, scroll down to the bottom and select BCPA CTA
* On the Pattern Assembler screen
  * Select any color or font variation
  * Click on `Unlock this style`
  * Ensure the copy is updated as mentioned above
  * Click on `Upgrade later`
  * Ensure the modal is close
  * Ensure the copy at the bottom of the sidebar changes to `Continue`
  * Ensure the copy above that button changes as mentioned above
  * Click on `Continue`, and ensure you're going to the site editor directly
* Test that the modal in other flows (in Design Picker themes, Showcase themes, or the Editor) continue to show the previous text.

## Pre-merge Checklist
